apps/google-analytics-4/frontend/src/components/config-screen/assign-content-type/AssignContentTypeCard.tsx

```diff
@@ -0,0 +1,136 @@
+import {
+  Box,
+  Button,
+  Card,
+  FormControl,
+  Select,
+  Stack,
+  TextInput,
+  TextLink,
+} from '@contentful/f36-components';
+import { css } from 'emotion';
+import { AllContentTypes, ContentTypeEntries } from 'types';
+
+interface Props {
+  allContentTypes: AllContentTypes;
+  contentTypeEntries: ContentTypeEntries;
+  onContentTypeChange: (prevKey: string, newKey: string) => void;
+  onContentTypeFieldChange: (key: string, field: string, value: string) => void;
+  onAddContentType: () => void;
+  onRemoveContentType: (key: string) => void;
+}
+
+const styles = {
+  contentTypeItem: css({
+    flex: 4,
+  }),
+  removeItem: css({
+    flex: 1,
+  }),
+};
+
+const AssignContentTypeCard = (props: Props) => {
+  const {
+    allContentTypes,
+    contentTypeEntries,
+    onContentTypeChange,
+    onContentTypeFieldChange,
+    onAddContentType,
+    onRemoveContentType,
+  } = props;
+
+  return (
+    <Card>
+      {Object.keys(contentTypeEntries).length ? (
+        <Stack marginBottom="none" spacing="spacingXs">
+          <Box className={styles.contentTypeItem}>
+            <FormControl marginBottom="none">
+              <FormControl.Label>Content type</FormControl.Label>
+            </FormControl>
+          </Box>
+          <Box className={styles.contentTypeItem}>
+            <FormControl marginBottom="none">
+              <FormControl.Label>Slug field</FormControl.Label>
+            </FormControl>
+          </Box>
+          <Box className={styles.contentTypeItem}>
+            <FormControl marginBottom="none">
+              <FormControl.Label>URL prefix</FormControl.Label>
+            </FormControl>
+          </Box>
+          <Box className={styles.removeItem}></Box>
+        </Stack>
+      ) : null}
+      {Object.entries(contentTypeEntries).map(([key, { slugField, urlPrefix }], index) => {
+        return (
+          <Stack spacing="spacingXs" paddingBottom="spacingS" key={key}>
+            <Box className={styles.contentTypeItem}>
+              <Select
+                id={`contentType-${index}`}
+                name={`contentType-${index}`}
+                onChange={(event: React.ChangeEvent<HTMLSelectElement>) =>
+                  onContentTypeChange(key, event.target.value)
+                }
+                value={key}>
+                <Select.Option value="" isDisabled>
+                  Select content type
+                </Select.Option>
+                {Object.entries(allContentTypes)
+                  .filter(([type]) => type === key || !contentTypeEntries[type])
+                  .map(([type, { name: typeName }]) => {
+                    return (
+                      <Select.Option value={type} key={`type-${type}`}>
+                        {typeName}
+                      </Select.Option>
+                    );
+                  })}
+              </Select>
+            </Box>
+            <Box className={styles.contentTypeItem}>
```

Comment:
These dropdowns is the example where I think refactoring here isn't exactly low hanging fruit but could be worth it if you find something elegant.

Reply:
Agreed that this will take bit more effort, so I am going to leave it as is for now. I have refactored the header labels to reduce repeatability.